src/StreamJsonRpc.Shared/Reflection/TargetMethod.cs

```diff
@@ -28,12 +32,43 @@ internal TargetMethod(JsonRpcMessage request, object target, JsonSerializer json
             this.target = target;
 
             var targetMethods = new Dictionary<MethodSignature, object[]>();
+            bool errorFound = false;
             for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
             {
-                bool matchFound = false;
-                foreach (MethodInfo method in t.GetDeclaredMethods(request.Method))
+                var matchFound = false;
+                var methodName = request.Method;
+                var methodAttribute = request.Method;
+
+                if (methodToAttributeMap != null)
+                {
+                    foreach (string methodKey in methodToAttributeMap.Keys)
```

Comment:
You're looking up the value for this key a few times in this loop. That's extra dictionary CPU time that you don't need if you enumerate the dictionary instead of its Keys collection.